pkg/providers/subnet/subnet.go

```diff
@@ -83,6 +83,9 @@ func (p *Provider) List(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTempl
 	if p.cm.HasChanged("subnets", subnetLog) {
 		logging.FromContext(ctx).With("subnets", subnetLog).Debugf("discovered subnets")
 	}
+	if len(output.Subnets) == 0 {
+		logging.FromContext(ctx).Errorf("no subnets exist given constraint")
```

Comment:
Does it make sense to log an error instead of returning an error? Also, did you make sure that this and the security groups act the same as the AWSCLI? E.g. If you do a `aws ec2 describe-subnets` call that returns no subnets, does AWS consider it an error? 

Overall, if you test this with subnets and security groups values that are nil and find that everything's fine, then it's all fine :D

Reply:
Just checked, AWSCLI doesn't error. It returns an empty array